delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/api/JsonLdJavaApi.scala

```diff
@@ -0,0 +1,150 @@
+package ch.epfl.bluebrain.nexus.delta.rdf.jsonld.api
+import akka.http.scaladsl.model.Uri
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.rdf.implicits._
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.IOErrorOr
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.JsonLdError._
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context._
+import com.github.jsonldjava.core.{Context, DocumentLoader, JsonLdProcessor, JsonLdOptions => JsonLdJavaOptions}
+import com.github.jsonldjava.utils.JsonUtils
+import io.circe.{parser, Json}
+import monix.bio.IO
+import org.apache.jena.iri.IRI
+import org.apache.jena.rdf.model.{Model, ModelFactory}
+import org.apache.jena.riot.system.StreamRDFLib
+import org.apache.jena.riot.{Lang, RDFParser, RDFWriter}
+
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+/**
+  * Json-LD high level API implementation by Json-LD Java library
+  */
+object JsonLdJavaApi extends JsonLdApi {
+
+  System.setProperty(DocumentLoader.DISALLOW_REMOTE_CONTEXT_LOADING, "true")
+
+  // $COVERAGE-OFF$
+  override def compact[Ctx <: JsonLdContext](input: Json, ctx: Json, f: ContextFields[Ctx])(implicit
+      opts: JsonLdOptions,
+      resolution: RemoteContextResolution
+  ): IOErrorOr[(Json, Ctx)] =
+    for {
+      obj           <- catchTryErr(JsonUtils.fromString(input.noSpaces), "building input")
+      ctxObj        <- catchTryErr(JsonUtils.fromString(ctx.noSpaces), "building context")
+      options       <- remoteContextLoader(input, ctx).map(toOpts)
+      compacted     <- catchTryErr(JsonUtils.toString(JsonLdProcessor.compact(obj, ctxObj, options)), "compacting")
+      compactedJson <- IO.fromEither(parser.parse(compacted).leftMap(err => UnexpectedJsonLd(err.getMessage())))
+      ctxValue      <- context(ctx, f, options)
+    } yield (compactedJson, ctxValue)
+  // $COVERAGE-ON$
+
+  override def expand(
+      input: Json
+  )(implicit options: JsonLdOptions, resolution: RemoteContextResolution): IOErrorOr[Json] =
+    for {
+      obj          <- catchTryErr(JsonUtils.fromString(input.noSpaces), "building input")
+      options      <- remoteContextLoader(input).map(toOpts)
+      expanded     <- catchTryErr(JsonUtils.toString(JsonLdProcessor.expand(obj, options)), "expanding")
+      expandedJson <- IO.fromEither(parser.parse(expanded).leftMap(err => UnexpectedJsonLd(err.getMessage())))
+    } yield expandedJson
+
+  override def frame[Ctx <: JsonLdContext](input: Json, frame: Json, f: ContextFields[Ctx])(implicit
```

Comment:
frame, expand and compact have common steps, maybe we can put do something with it

Reply:
It is better understood like this than having a obscure refactoring. Repetition is not always a bad thing.

Reply:
Agreed.

What about something like:
`private def process(input: Json, stage: Stage)(f: Any => Any) = // Stages could typed too
    for {
      obj             <- fromString(input.noSpaces, "building input")
      options      <- remoteContextLoader(input).map(toOpts)
      result         <- toJson(f(obj, options)), stage)
    } yield result

def expand(input: Json) = process(input, Stage.Expanding)(JsonLdProcessor.expand) 
`
JsonLdProcessor makes the thing a little meh

Reply:
I've try few things like that, but they introduce more code and more jumping from here to there.

Since this is the Json-LD Java implementation part of the code I think it is clear and straightforward as is. If it evolves to something else in the future, we can consider refactoring.